commons/src/main/java/com/pryv/model/Access.java

```diff
@@ -0,0 +1,332 @@
+   *
+   * @param result The ResultSet
+   * @return The Acccess
+   * @throws SQLException
+   * @throws IOException
+   */
+  public static Access createOrReuse(ResultSet result) throws SQLException, IOException {
+    // TODO: Add query keys
+    String id = result.getString(QueryGenerator.ACCESSES_ID_KEY);
+    Access access = supervisor.get(id);
+    if (access == null) {
+      access = new Access();
+    }
+
+    access.setId(result.getString(QueryGenerator.ACCESSES_ID_KEY));
+    access.setToken(result.getString(QueryGenerator.ACCESSES_TOKEN_KEY));
+    access.setName(result.getString(QueryGenerator.ACCESSES_NAME_KEY));
+    // TODO: Deserialize and set permissions
+    access.setCreated(result.getDouble(QueryGenerator.ACCESSES_CREATED_KEY));
+    access.setCreatedBy(result.getString(QueryGenerator.ACCESSES_CREATED_BY_KEY));
+    access.setModified(result.getDouble(QueryGenerator.ACCESSES_MODIFIED_KEY));
+    access.setModifiedBy(result.getString(QueryGenerator.ACCESSES_MODIFIED_BY_KEY));
+    access.setType(result.getString(QueryGenerator.ACCESSES_TYPE_KEY));
+    access.setDeviceName(result.getString(QueryGenerator.ACCESSES_DEVICE_NAME_KEY));
+    access.setLastUsed(result.getDouble(QueryGenerator.ACCESSES_LAST_USED_KEY));
+
+    return access;
+  }
+
+  /**
+   * saves the Access in the supervisor if needed
+   *
+   * @return the Access
+   */
+  public static Access createOrReuse(Access access) {
+    String id = access.getId();
+    // TODO: merge - not replace
+    supervisor.put(id, access);
+    return access;
+  }
+
+  /**
+   * Assign unique identifier to the Access - does nothing if Access has already a id field
+   */
+  private String generateId() {
+    if (this.id == null) {
+      this.id = Cuid.createCuid();
+    }
+    return this.id;
+  }
+
+  private void updateSupervisor() {
+    String id = this.getId();
+    if(supervisor.containsKey(id)) {
+      supervisor.get(id).merge(this, JsonConverter.getCloner());
+    } else {
+      supervisor.put(id,this);
+    }
+  }
+
+  /**
+   * Assign a weak reference to the ConnectionOld
+   *
+   * @param weakconnection
+   */
+  public void assignConnection(WeakReference<AbstractConnection> weakconnection) {
+    this.weakConnection = weakconnection;
+  }
+
+  /**
+   * Returns the reference to the ConnectionOld to which the Access is linked if
+   * any.
+   *
+   * @return
+   */
+  public AbstractConnection getWeakConnection() {
+    return weakConnection.get();
+  }
+
+  /**
+   * Copy all temp Access's values into caller Access.
+   *
+   * @param temp
+   *          the Access from which the fields are merged
+   * @param cloner
+   *          com.rits.cloning.Cloner instance from JsonConverter util class
+   */
+  public void merge(Access temp, Cloner cloner) {
+    weakConnection = temp.weakConnection;
+
+    id = temp.id;
+    token = temp.token;
+    name = temp.name;
+    for (Permission permission : temp.permissions) {
```

Comment:
I think that you should do a deep copy here

Reply:
Ok